platform-specific/dtls_prng_posix.c

```diff
@@ -18,6 +18,12 @@
  *
  *******************************************************************************/
 
+#ifdef HAVE_RANDOM
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
```

Comment:
Strange, why is "random()" found, if it requires _GNU_SOURCE?

I'm not sure, of one of the other solutions in [stackoverflow](https://stackoverflow.com/questions/5582211/what-does-define-gnu-source-imply) would have less side-effects.

Reply:
It does not have to be _GNU_SOURCE, but in <stdlib.h> random() is wrapped by
````
#if defined __USE_MISC || defined __USE_XOPEN_EXTENDED
````
or I have in another place (Centos 6)
````
#if defined __USE_SVID || defined __USE_XOPEN_EXTENDED || defined __USE_BSD
````
both of which respond to _GNU_SOURCE being set.

Trying `#define _POSIX_C_SOURCE 200809L`  makes no difference, but `#define _XOPEN_SOURCE 700` does work.  And, `_GNU_SOURCE `is only getting defined in `platform-specific/dtls_prng_posix.`c .